File: Golang/Transpose Matrix.go
func transpose(matrix [][]int) [][]int {
  Y, X := len(matrix), len(matrix[0])

  M := make([][]int, X)
  for x := 0; x < X; x++ {
    M[x] = make([]int, Y)
    for y := 0; y < Y; y++ {
      M[x][y] = matrix[y][x]
    }
  }

  return M
}

File: Python3/Transpose Matrix.py
#1
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        res = []
        for column in range(len(matrix[0])):
            temp = []
            for row in range(len(matrix)):
                temp.append(matrix[row][column])
            res.append(temp)

        return res

#2
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        rows, cols = len(matrix), len(matrix[0])
        result = [[0 for _ in range(rows)] for _ in range(cols)]
        for i in range(rows):
            for j in range(cols):
                result[j][i] = matrix[i][j]
        return result